src/agent/v8/v8inspector.ts

```diff
@@ -92,4 +142,63 @@ export class V8Inspector {
     );
     return result;
   }
+
+  /** Attaches to the V8 debugger. */
+  private attach() {
+    if (this.session) {
+      return;
+    }
+
+    const session = new inspector.Session();
+    session.connect();
+    session.on('Debugger.scriptParsed', script => {
+      this.onScriptParsed(script.params);
+    });
+    session.post('Debugger.enable');
+    session.post('Debugger.setBreakpointsActive', {active: true});
+    session.on('Debugger.paused', message => {
+      this.onPaused(message.params);
+      this.resetV8DebuggerIfMeetThreshold();
+    });
+
+    this.session = session;
+  }
+
+  /** Detaches from the V8 debugger. */
```

Comment:
Please add a note that it purges breakpoint state (ie. detach + attach is lossy)

Reply:
Done.